genePred/compare.go

```diff
@@ -1,61 +1,65 @@
 package genePred
 
 import (
-	"log"
 	"strings"
 )
 
-func AllAreEqual(a []*GenePred, b []*GenePred) bool {
+func AllAreEqual(a []GenePred, b []GenePred) (bool, []int) {
+	var fields []int
+	var equal bool
+
 	if len(a) != len(b) {
-		return false
+		return false, fields
 	}
 	for i := 0; i < len(a); i++ {
-		if !Equal(a[i], b[i]) {
-			return false
-		}
+		equal, fields = Equal(a[i], b[i])
+		return equal, fields
 	}
-	return true
+	return equal, fields
 }
 
-func Equal(a *GenePred, b *GenePred) bool {
+func Equal(a GenePred, b GenePred) (bool, []int) {
+	var isEqual = true
+	var fields = make([]int, 9)
```

Comment:
I see that down below you access fields[9].  Because the slice starts at zero, you need it to have a length of 10 to be able to access fields[9].

Reply:
Is the vision that this function will be called many times and the records will usually match?  If so, we are going to have a constant allocation and non-use of this slice.  Another option would be to declare an empty / nil slice here: "var fields []int" and then use append() to add all the non-matching fields to it (when needed).  For example, instead of "fields[2] = 1", you would have fields = append(fields, 2)".  You would then return something like "(2,4,5)" instead of "(0, 0, 1, 0, 1, 1)".  Another thing to think about is if []bool would be a better fit than []int if you like the current style since it seems like the int is really holding a 0/1 false/true

Reply:
i went with the idea that the field that had the error is stored as an int in the slice. Thank you for pointing out these options!